### PR TITLE
Make prettyprint’s cycle detection aware of Delegator instances. Fixes #13144.

### PR DESCRIPTION
**Pull request by richard-viney**
https://bugs.ruby-lang.org/issues/13144

### CODE REVIEW

**Comment by shyouhei** [comment]
@nobu can you take a look?

File: lib/pp.rb [hunk 1]
```diff
@@ -149,6 +150,10 @@ def pop_inspect_key(id)
     # Object#pretty_print_cycle is used when +obj+ is already
     # printed, a.k.a the object reference chain has a cycle.
     def pp(obj)
+      # If obj is a Delegator then use the object being delegated to for cycle
+      # detection
+      obj = obj.__getobj__ if obj.is_a? Delegator
```
Comment (nobu) [inline lib/pp.rb]: `defined?(::Delegator) and obj.is_a?(::Delegator)` would be better than requiring 'delegate' always, I think.

Comment (richard-viney) [reply lib/pp.rb]: Shall I make this change?

Comment (richard-viney) [reply lib/pp.rb]: I see there are also now conflicts on this PR, can have a look if that'd be useful.

**Pull request merged**